isocroft-syntax.md

```diff
@@ -0,0 +1,913 @@
+## SOURCE FILE
+- Source file can be empty, otherwise it must contain a valid set of (any 66536) UTF-8 codepoints.
+
+- Support for UTF-8 identifiers is planned.
+
+
+## COMMENTS
+#### Single-line comments
+- Single-line comments start with `#` and `//`
+    ```php
+    # Hello, World!
+    ```
+    ```java
+    // Hello, World!
+    ```
+
+    The singline comment starts with `#` so the plain source can be executed natively using shebang `(#!)` or system command can be executed at the start of the source file.
+
+#### Multiline comments
+- Nested multiline comment is allowed.
+    ```rust
+    /*
+     * Multiline comments can span multiple lines.
+     * They begin with `/*`.
+     * And they end with `*/`.
+     */
+    ```
+## DATA TYPES
+- Data types in ratio
+
+* int > a sequence of symbols from the standard number set (e.g. 2489)
+* long > two or more `int` type(s) put together as a sequence (e.g. 1267383994747489948947333344747)
+* float >
+* char > a single symbol from the standard chracter set ('$')
+* str > two or more `char` type(s) put together ('$rw32')
+* bool > true, false
+* byte > |01011010, |10010011
+
+## SUBJECT
+- Subject consist of a subject name and value
+
+    ```js
+    const num = 2561 // implicit definition as 4 bytes of int type for 2561
+    const int num = 2561 // explicit definition as 4 bytes of int type for 2561
+    
+    ```
+
+    ```js
+    const name = "John" // implicit definition as 4 bytes of char type for "John"
+    const char<4> name = "John" // explicit definition as 4 bytes of char type for "John" (a tuple of characters)
+    
+    // a string in ratio is therefore a tuple of characters which are immutable
+    const str name = "John"
+    ```
+
+    Here `name` is the name of the subject and `"John"` the value or object.
+
+- The `const` keyword is used for subject which value cannot change.
+    ```cpp
+    const name = "John"
+    name = "Doe" // Invalid!
+    ```
```

Comment:
i think the case you present for `let` OR `val` is quite subjective.  Maybe because you prefer or are more used to `let` than `const`. Nonetheless,  `const` (i believe) is more descriptive of the use. However, whatever we choose for Ratio will have to be debated more on concrete and objective reason than subjective preference.

Reply:
*`let` is used by several modern languages* is a good enough objective reason to choose it. It is true that I prefer `let` over `const` but if that is what the majority wants, then we go with it.